IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -506,7 +565,119 @@ - (BOOL)removeItemsWithTokenKey:(__unused MSIDCacheKey *)key
+    
+    if (sharedCredential)
+    {
+        [sharedCredential removeSharedCredentialForKey:(MSIDDefaultCredentialCacheKey *)key];
+        return [self saveSharedCredential:sharedCredential key:key context:context error:error];
+    }
+    
+    return YES;
+}
+
+- (BOOL)removeAppCredentialsWithTokenKey:(__unused MSIDCacheKey *)key
+                                 context:(__unused id<MSIDRequestContext>)context
+                                   error:(__unused NSError **)error
+{
+    MSIDMacAppCredentialCacheItem *appCredential = [self appCredentialWithKey:key context:context error:error];
+    
+    if (appCredential)
+    {
+        [appCredential removeAppCredentialForKey:(MSIDDefaultCredentialCacheKey *)key];
+        return [self saveAppCredential:appCredential key:key context:context error:error];
+    }
+    
+    return YES;
+}
+
+- (MSIDMacSharedCredentialCacheItem *)sharedCredentialWithKey:(MSIDCacheKey *)key
+                                                      context:(id<MSIDRequestContext>)context
+                                                        error:(NSError **)error
+{
+    MSID_TRACE;
+    MSIDCacheItemJsonSerializer *serializer = [[MSIDCacheItemJsonSerializer alloc] init];
+    MSIDMacSharedCredentialCacheItem *sharedCredentialItem = nil;
+    NSMutableDictionary *query = [self.defaultCacheQuery mutableCopy];
+    query[(id)kSecAttrAccount] = self.keychainGroup;
+    query[(id)kSecAttrService] = s_defaultKeychainLabel;
+    query[(id)kSecReturnAttributes] = (__bridge id)kCFBooleanTrue;
+    query[(id)kSecReturnData] = (__bridge id)kCFBooleanTrue;
+    
+    CFDictionaryRef result = nil;
+    OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query, (CFTypeRef *)&result);
+    if (status == errSecSuccess)
+    {
+        NSDictionary *resultDict = (__bridge_transfer NSDictionary *)result;
+        NSData *sharedCredentialData = [resultDict objectForKey:(id)kSecValueData];
+        sharedCredentialItem = (MSIDMacSharedCredentialCacheItem *)[serializer deserializeSharedCredentialCacheItem:sharedCredentialData];
+        
+        if (!sharedCredentialItem)
+        {
+            MSID_LOG_WARN(context, @"Failed to deserialize shared token");
+        }
+    }
+    
+    return sharedCredentialItem;
+}
+
+- (MSIDMacAppCredentialCacheItem *)appCredentialWithKey:(MSIDCacheKey *)key
+                                                context:(id<MSIDRequestContext>)context
+                                                  error:(NSError **)error
+{
+    MSID_TRACE;
+    MSIDCacheItemJsonSerializer *serializer = [[MSIDCacheItemJsonSerializer alloc] init];
+    MSIDMacAppCredentialCacheItem *appCredentialItem = nil;
+    NSMutableDictionary *query = [self.defaultCacheQuery mutableCopy];
+    query[(id)kSecAttrAccount] = [NSString stringWithFormat:@"%@-%@", self.keychainGroup,[[NSBundle mainBundle] bundleIdentifier]];;
```

Comment:
```suggestion
    query[(id)kSecAttrAccount] = [NSString stringWithFormat:@"%@-%@", self.keychainGroup,[[NSBundle mainBundle] bundleIdentifier]];
```
Double-semicolons will cause a compiler warning which will break in msalcpp since it is configured to treat all warnings as errors.

Reply:
Can we make sure our Obj-C project settings are inline with CPP so we catch those issues?

Reply:
I don't know if there is xcode setting to find double semi-colon. I think we can run some custom util from the command line to catch all cases. For now, I have removed all instances of double ;; from the source code.

Reply:
I'm probably thinking of a different semicolon-related warning. I'll check.

Reply:
I added a task to get the compiler flags more in-sync. For now, I think we can continue to just fix things as-needed.